### ISSUE TITLE
gNMI southbound should send attribute values in the format defined in the model (not just string)

### ISSUE THREAD

**Issue by SeanCondon** [onosproject/onos-config#303]
To date when setting attributes on the gNMI southbound we have always used the string_val type assuming the target device will know how to cast it in to the right data type.

In the case of DeviceSim this is not the case. e.g. setting the value of enabled with:
```
gnmi_cli -address localhost:5150 -set     -proto "update: <path: <target: 'localhost:10161', elem: <name: 'interfaces'> elem:<name:'interface' key: <key: 'name' value: 'eth1'>> elem: <name: 'config'> elem: <name: 'enabled'>> val: <bool_val: false>>"     -timeout 5s -alsologtostderr     -client_crt pkg/southbound/testdata/client1.crt     -client_key pkg/southbound/testdata/client1.key     -ca_crt pkg/southbound/testdata/onfca.crt
```

gives
```
2019/06/05 13:11:49 SetResponse  target returned RPC error for Set("update:<path:<elem:<name:\"interfaces\" > elem:<name:\"interface\" key:<key:\"name\" value:\"eth1\" > > elem:<name:\"config\" > elem:<name:\"enabled\" > > val:<string_val:\"false\" > > ") : rpc error: code = Internal desc = error in creating config struct from IETF JSON data: got string type for field enabled, expect bool

```

**Comment by SeanCondon** [comment]
There are 2 options for solving this:

1. Keep the type of attribute with the value when we store it. This has the disadvantage that the user would have to know the data type to use when setting a value. Also there is a risk the user could specify a different type each time.
2. Allow the integration with YGOT to look up the type from the model and assign the correct type. This would have to happen for configs sent southbound and for GetResponses northbound

**Comment by Andrea-Campanella** [comment]
in option 1 you can use ygot at the time the user asks for storing something, in case 2 you are using it when it's stored to a device.

**Comment by SeanCondon** [comment]
Closed by #491
